portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,21 +37,19 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long maxComments = Long.parseLong(request.getParameter("maxComments"));
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));
```

Comment:
What happens if the parameter doesn't exist or can't be parsed to an integer?

Reply:
the servlet will just quit and the client will timeout. added improved error handling now. https://github.com/thomasogara/step-repository/pull/37/commits/d31f2e907d0c77d8c6b5d985e0468fa737bf17e0

Reply:
I used the constant ALL_COMMENTS just to make it clearer what -1 represents in this context